src/components/CheckboxWithLabel.tsx

```diff
@@ -0,0 +1,108 @@
+import React, {ComponentType, ForwardedRef, useState} from 'react';
+import {StyleProp, View, ViewStyle} from 'react-native';
+import useThemeStyles from '@styles/useThemeStyles';
+import variables from '@styles/variables';
+import Checkbox from './Checkbox';
+import FormHelpMessage from './FormHelpMessage';
+import PressableWithFeedback from './Pressable/PressableWithFeedback';
+import Text from './Text';
+
+type RequiredLabelProps =
+    | {
+          /** Text that appears next to check box */
+          label: string;
+
+          /** Component to display for label
+           * If label is provided, LabelComponent is not required
+           */
+          LabelComponent?: ComponentType;
+      }
+    | {
+          /** Component to display for label */
+          LabelComponent: ComponentType;
+
+          /** Text that appears next to check box
+           * If LabelComponent is provided, label is not required
+           */
+          label?: string;
+      };
+
+type CheckboxWithLabelProps = RequiredLabelProps & {
+    /** Whether the checkbox is checked */
+    isChecked?: boolean;
+
+    /** Called when the checkbox or label is pressed */
+    onInputChange?: (value?: boolean) => void;
+
+    /** Container styles */
+    style?: StyleProp<ViewStyle>;
+
+    /** Error text to display */
+    errorText?: string;
+
+    /** Value for checkbox. This prop is intended to be set by Form.js only */
+    value?: boolean;
+
+    /** The default value for the checkbox */
+    defaultValue?: boolean;
+
+    /** The ID used to uniquely identify the input in a Form */
+    /* eslint-disable-next-line react/no-unused-prop-types */
+    inputID?: string;
+
+    /** Saves a draft of the input value when used in a form */
+    // eslint-disable-next-line react/no-unused-prop-types
+    shouldSaveDraft?: boolean;
```

Comment:
Why do we need this one?

Reply:
I am not sure, but it was there before I started migrating it and it had that eslint disable line 😅 do you think we should remove it?

Reply:
It's necessary when used inside `Form` component, same as `inputID`.